src/toil/lib/aws/iam.py

```diff
@@ -0,0 +1,140 @@
+        raise RuntimeError("Missing permissions", permissions)
+
+    return None
+
+
+def helper_permission_check(perm: str, list_perms: List[str]) -> bool:
+    """
+    Takes a permission and checks whether it's allowed against a list of allowed permissions
+
+    :param perm: Permission to check in string form
+    :param list_perms: Permission list to check against
+    """
+    flag = False
+    for allowed in list_perms:
+        if allowed[0] == "*":
+            if perm.endswith(allowed[1:]):
+                flag = True
+
+        if allowed[0] == "*" and allowed[-1] == "*":
+            if allowed[1:-1] in perm:
+                flag = True
+
+        if allowed[-1] == "*":
+            if perm.startswith(allowed[:-1]):
+                flag = True
+
+        if allowed == perm:
+            flag = True
+    if not flag:
+
+        return False
+    else:
+        return True
+
+
+
+def test_dummy_perms() -> bool:
+    """
+    Test for success of check policy warning against dummy permissions
+    """
+    launch_tester = {'*': ['ec2:*', 'iam:*', 's3:*', 'sdb:*']}
+
+    check_policy_warnings(allowed_actions=launch_tester)
+    print("Success")
+    return True
+
+
+def get_allowed_actions() -> Dict[str, List[str]]:
+    """
+    Returns a list of all allowed actions in a dictionary which is keyed by resource permissions
+    are allowed upon.
+    """
+    aws = AWSConnectionManager()
+
+    region = zone_to_region(get_best_aws_zone() or "us-west-2a" )
+
+
+    iam: IAMClient = cast(IAMClient, aws.client(region, 'iam'))
+
+    response = iam.get_instance_profile(InstanceProfileName="fakename_toil")
+
+    role_name = response['InstanceProfile']['Roles'][0]['RoleName']
+
+    list_policies = iam.list_role_policies(RoleName=role_name)
+
+    account_num = boto3.client('sts').get_caller_identity().get('Account')
+
+    str_arn = f"arn:aws:iam::{account_num}:role/{role_name}"
+
+    role_name = response['InstanceProfile']['Roles'][0]['RoleName']
+
+    list_policies = iam.list_role_policies(RoleName=role_name)
+
+    account_num = boto3.client('sts').get_caller_identity().get('Account')
+
+    allowed_actions: Dict[Any, Any] = {}
+
+    for policy_name in list_policies['PolicyNames']:
+        policy_arn = f"arn:aws:iam::{account_num}:policy/{policy_name}"
+
+        role_policy: Dict[Any, Any] = dict(iam.get_role_policy(
+            RoleName=role_name,
+            PolicyName=policy_name
+        ))
+
+        if role_policy["PolicyDocument"]["Statement"][0]["Effect"] == "Allow":
+            if role_policy["PolicyDocument"]["Statement"][0]["Resource"] not in allowed_actions.keys():
+                allowed_actions[role_policy["PolicyDocument"]["Statement"][0]["Resource"]] = []
+
+            allowed_actions[role_policy["PolicyDocument"]["Statement"][0]["Resource"]].append(
+                role_policy["PolicyDocument"]["Statement"][0]["Action"])
+
+    check_policy_warnings(allowed_actions)
```

Comment:
I don't think this call should be here; we just want to return the permission grants and we will check them against something later, right?